### PR TITLE
Replace OverlapReal with ShortReal.

### LINKED ISSUE #1447: Replace `OverlapReal` with `ShortReal`.
### Description

Global find and replace `OverlapReal` with `ShortReal`.

### Motivation and context

#1444 makes the mixed precision types available for the entire codebase, where `OverlapReal` is specific to HPMC. 

Note: To avoid merge conflicts, defer this task until just before the v4 release and preferably when there is no ongoing work on HPMC based on `trunk-minor`.

### PR DESCRIPTION
**Pull request by joaander**
## Description

* Replace all instances of `OverlapReal` with `ShortReal`.
* Remove the now unused `HPMCPrecisionSetup.h`.

## Motivation and context

`ShortReal` is the new mechanism for managing mixed precision builds. It replaces the old `OverlapReal` type which was limited to HPMC.

Resolves #1447.

## How has this been tested?

Local builds pass.

## Change log

```
Removed:

* ``hpmc::OverlapReal`` C++ data type. Use ``ShortReal``.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/hpmc/ShapeConvexPolygon.h [hunk 1]
```diff
@@ -387,14 +385,14 @@ struct ShapeConvexPolygon
...
-        // vec2<OverlapReal> s_y = rotate(o, sfunc(e_y));
-        // vec2<OverlapReal> s_neg_x = rotate(o, sfunc(-e_x));
-        // vec2<OverlapReal> s_neg_y = rotate(o, sfunc(-e_y));
+        // vec2<ShortReal> s_x = rotate(o, sfunc(e_x));
+        // vec2<ShortReal> s_y = rotate(o, sfunc(e_y));
+        // vec2<ShortReal> s_neg_x = rotate(o, sfunc(-e_x));
+        // vec2<ShortReal> s_neg_y = rotate(o, sfunc(-e_y));
 
         // // translate out from the position by the furthest extents
         // vec3<Scalar> upper(pos.x + s_x.x, pos.y + s_y.y, 0.1);
```
Comment (b-butler) [inline hoomd/hpmc/ShapeConvexPolygon.h]: Is there a reason we keep this?

File: hoomd/hpmc/ShapeSpheropolyhedron.h [hunk 2]
```diff
@@ -112,16 +112,16 @@ struct ShapeSpheropolyhedron
...
+        // quat<ShortReal> o(orientation);
+        // vec3<ShortReal> e_x(1,0,0);
+        // vec3<ShortReal> e_y(0,1,0);
+        // vec3<ShortReal> e_z(0,0,1);
+        // vec3<ShortReal> s_x = rotate(o, sfunc(rotate(conj(o),e_x)));
+        // vec3<ShortReal> s_y = rotate(o, sfunc(rotate(conj(o),e_y)));
+        // vec3<ShortReal> s_z = rotate(o, sfunc(rotate(conj(o),e_z)));
+        // vec3<ShortReal> s_neg_x = rotate(o, sfunc(rotate(conj(o),-e_x)));
+        // vec3<ShortReal> s_neg_y = rotate(o, sfunc(rotate(conj(o),-e_y)));
+        // vec3<ShortReal> s_neg_z = rotate(o, sfunc(rotate(conj(o),-e_z)));
```
Comment (b-butler) [inline hoomd/hpmc/ShapeSpheropolyhedron.h]: Same comment as before.

File: hoomd/hpmc/test/test_convex_polygon.cc [hunk 3]
```diff
@@ -69,31 +69,31 @@ PolygonVertices setup_verts(const vector<vec2<OverlapReal>> vlist)
...
 //         {
-//         vec2<OverlapReal> line = ss_verts.v[i+1] - ss_verts.v[i];
+//         vec2<ShortReal> line = ss_verts.v[i+1] - ss_verts.v[i];
 //         o << "text " << i << endl;
 //         o << "rline " << line.x << " " << line.y << endl;
 //         }
 
 //     o << "text " << ss_verts.N-1 << endl;
-//     vec2<OverlapReal> line = ss_verts.v[0] - ss_verts.v[ss_verts.N-1];
+//     vec2<ShortReal> line = ss_verts.v[0] - ss_verts.v[ss_verts.N-1];
```
Comment (b-butler) [inline hoomd/hpmc/test/test_convex_polygon.cc]: Why this comment?

File: hoomd/hpmc/test/test_convex_polyhedron.cc [hunk 4]
```diff
@@ -90,7 +90,7 @@ UP_TEST( composite_support )
     // Find the support of the Minkowski difference of two offset cubes
     quat<Scalar> o;
 
-    vector< vec3<OverlapReal> > vlist;
+    vector< vec3<ShortReal> > vlist;
```
Comment (b-butler) [inline hoomd/hpmc/test/test_convex_polyhedron.cc]: More commented code.

File: hoomd/hpmc/test/test_simple_polygon.cc [hunk 5]
```diff
@@ -58,33 +58,33 @@ PolygonVertices setup_verts(const vector<vec2<OverlapReal>> vlist)
     return result;
     }
 
-// void write_gle_header(ostream& o, OverlapReal s)
+// void write_gle_header(ostream& o, ShortReal s)
```
Comment (b-butler) [inline hoomd/hpmc/test/test_simple_polygon.cc]: Not even used commented function.

File: hoomd/hpmc/test/test_spheropolyhedron.cc [hunk 6]
```diff
@@ -149,7 +149,7 @@ UP_TEST( composite_support )
     // Find the support of the Minkowski difference of two offset cubes
     quat<Scalar> o;
 
-    vector< vec3<OverlapReal> > vlist;
+    vector< vec3<ShortReal> > vlist;
```
Comment (b-butler) [inline hoomd/hpmc/test/test_spheropolyhedron.cc]: More commented code.

File: hoomd/hpmc/ShapeConvexPolyhedron.h [hunk 7]
```diff
@@ -751,16 +751,16 @@ struct ShapeConvexPolyhedron
...
-        // vec3<OverlapReal> e_x(1,0,0);
-        // vec3<OverlapReal> e_y(0,1,0);
-        // vec3<OverlapReal> e_z(0,0,1);
-        // vec3<OverlapReal> s_x = rotate(o, sfunc(rotate(conj(o),e_x)));
-        // vec3<OverlapReal> s_y = rotate(o, sfunc(rotate(conj(o),e_y)));
-        // vec3<OverlapReal> s_z = rotate(o, sfunc(rotate(conj(o),e_z)));
-        // vec3<OverlapReal> s_neg_x = rotate(o, sfunc(rotate(conj(o),-e_x)));
-        // vec3<OverlapReal> s_neg_y = rotate(o, sfunc(rotate(conj(o),-e_y)));
-        // vec3<OverlapReal> s_neg_z = rotate(o, sfunc(rotate(conj(o),-e_z)));
+        // quat<ShortReal> o(orientation);
```
Comment (b-butler) [inline hoomd/hpmc/ShapeConvexPolyhedron.h]: Commented out code.

File: hoomd/hpmc/ShapeSphinx.h [hunk 8]
```diff
@@ -1102,7 +1101,7 @@ DEVICE inline bool test_overlap<ShapeSphinx, ShapeSphinx>(const vec3<Scalar>& r_
         }
     /*if((p.n == 5) && (q.n == 4))
```
Comment (b-butler) [inline hoomd/hpmc/ShapeSphinx.h:1102]: More commented code.

File: hoomd/hpmc/SphinxOverlap.h [hunk 9]
```diff
@@ -3337,17 +3334,17 @@ DEVICE inline bool sep8(bool convex,
     return false;
     }
 
-/*  OverlapReal op3(OverlapReal ab,OverlapReal ac,OverlapReal ad,
-               OverlapReal bc,OverlapReal bd,
-               OverlapReal cd)
+/*  ShortReal op3(ShortReal ab,ShortReal ac,ShortReal ad,
```
Comment (b-butler) [inline hoomd/hpmc/SphinxOverlap.h:3337]: Commented code.

**Review by b-butler — APPROVED** [review]
Looks good a simple find and replace for the most part. I left comments where I found changes in commented code; there was quite a bit. I don't want to try and hold up this PR for that, so if you feel like removing it before merging you can, but otherwise :+1: .

**Review by cbkerr — APPROVED** [review]
:)

**Comment by joaander** [comment]
The commented code fell into two categories: 1) Alternate implementations of methods that ended up being slower when benchmarked or 2) Methods that wrote out visual information from unit tests so that the shape overlap checks could be inspected visually. I removed these.

**Pull request merged**